packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/ViewToolSectionsProvider.java

```diff
@@ -68,44 +105,188 @@ public List<ToolSection> handle(Object targetElement, Object diagramElement, Obj
     }
 
-    private Optional<org.eclipse.sirius.components.view.DiagramDescription> getSourceDiagramDescription(DiagramDescription diagramDescription) {
-        if (diagramDescription != null && diagramDescription.getCanCreatePredicate() instanceof IViewDiagramCreationPredicate viewCreationPredicate) {
-            return Optional.of(viewCreationPredicate.getSourceDiagramDescription());
-        } else {
-            return Optional.empty();
-        }
-    }
-
-    private Optional<ToolSection> findToolSectionById(DiagramDescription diagramDescription, String id) {
-        return diagramDescription.getToolSections().stream().filter(section -> Objects.equals(section.getId(), id)).findFirst();
-    }
-
     private List<ToolSection> getDiagramPalette(DiagramDescription diagramDescription) {
         String sourceElementId = this.getSourceElementId(diagramDescription.getId());
+        // Palette for the diagram itself
         String diagramPaletteId = "siriusComponents://diagramPalette?diagramId=" + sourceElementId;
-        return this.findToolSectionById(diagramDescription, diagramPaletteId).stream().toList();
+        var allToolSections = new ArrayList<ToolSection>();
+        var optionalDiagramDescription = this.viewRepresentationDescriptionSearchService.findById(diagramDescription.getId())
+                .filter(org.eclipse.sirius.components.view.DiagramDescription.class::isInstance)
+                .map(org.eclipse.sirius.components.view.DiagramDescription.class::cast);
+
+        if (optionalDiagramDescription.isPresent()) {
+            org.eclipse.sirius.components.view.DiagramDescription viewDiagramDescription = optionalDiagramDescription.get();
+            viewDiagramDescription.getPalette();
+
+            // @formatter:off
+            var diagramPalette = ToolSection.newToolSection(diagramPaletteId)
+                    .label(viewDiagramDescription.getName())
+                    .tools(this.createDiagramPaletteTools(viewDiagramDescription))
+                    .imageURL("")
+                    .build();
+            // @formatter:on
+
+            allToolSections.add(diagramPalette);
+
+        }
+        return allToolSections;
+    }
+
+    private List<ITool> createDiagramPaletteTools(org.eclipse.sirius.components.view.DiagramDescription viewDiagramDescription) {
+        List<ITool> diagramTools = new ArrayList<>();
+        for (NodeTool nodeTool : viewDiagramDescription.getPalette().getNodeTools()) {
+            // @formatter:off
+            String toolId = this.idProvider.apply(nodeTool).toString();
+            var tool = SingleClickOnDiagramElementTool.newSingleClickOnDiagramElementTool(toolId)
+                    .label(nodeTool.getName())
+                    .imageURL(ViewToolImageProvider.NODE_CREATION_TOOL_ICON)
+                    //Empty handler
+                    .handler(variableManager -> {
+                        return null;
```

Comment:
Add a failure handler with the unexpected error message.
Log an error in the console as well